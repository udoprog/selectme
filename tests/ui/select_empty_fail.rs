pub(crate) fn main() {
}

/// Empty selects are not permitted.
async fn empty() {
    selectme::select! {}
}
