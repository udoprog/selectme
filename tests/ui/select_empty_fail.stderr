error: `select!` must not be empty, consider replacing with `future::pending::<()>().await` instead
 --> tests/ui/select_empty_fail.rs:6:5
  |
6 |     selectme::select! {}
  |     ^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `$crate::__support::select` which comes from the expansion of the macro `selectme::select` (in Nightly builds, run with -Z macro-backtrace for more info)
